storage/src/mmr/benches/update.rs

```diff
@@ -11,41 +11,57 @@ fn bench_update(c: &mut Criterion) {
+                            let mut h = Sha256::new();
+                            let mut h = Standard::new(&mut h);
 
-                        // Append random elements to MMR
-                        for _ in 0..leaves {
-                            let digest = sha256::Digest::random(&mut sampler);
-                            elements.push(digest);
-                            let pos = mmr.add(&mut h, &digest).await.unwrap();
-                            leaf_positions.push(pos);
-                        }
+                            // Append random elements to MMR
+                            for _ in 0..leaves {
+                                let digest = sha256::Digest::random(&mut sampler);
+                                elements.push(digest);
+                                let pos = mmr.add(&mut h, &digest).await.unwrap();
+                                leaf_positions.push(pos);
+                            }
 
-                        // Randomly update leaves -- this is what we are benchmarking.
-                        let start = Instant::now();
-                        for _ in 0..updates {
-                            let rand_leaf_num = sampler.gen_range(0..leaves);
-                            let rand_leaf_pos = leaf_positions[rand_leaf_num as usize];
-                            let rand_leaf_swap = sampler.gen_range(0..elements.len());
-                            let new_element = &elements[rand_leaf_swap];
-                            mmr.update_leaf(&mut h, rand_leaf_pos, new_element)
-                                .await
-                                .unwrap();
-                        }
-
-                        start.elapsed()
-                    });
-                },
-            );
+                            // Randomly update leaves -- this is what we are benchmarking.
+                            let start = Instant::now();
+                            for _ in 0..updates {
+                                let rand_leaf_num = sampler.gen_range(0..leaves);
+                                let rand_leaf_pos = leaf_positions[rand_leaf_num as usize];
+                                let rand_leaf_swap = sampler.gen_range(0..elements.len());
+                                let new_element = &elements[rand_leaf_swap];
+                                if batched {
+                                    mmr.update_leaf_batched(&mut h, rand_leaf_pos, new_element)
+                                        .await
+                                        .unwrap();
+                                    continue;
+                                }
+                                mmr.update_leaf(&mut h, rand_leaf_pos, new_element)
+                                    .await
+                                    .unwrap();
+                            }
+                            if batched {
+                                mmr.sync(&mut h);
```

Comment:
nit: I'm ok calling this always (seems like eventhough one is batched, they should both have to take the overhead of disk)?

Reply:
Removed the "if". It's a no-op if it's not batched, since this is only for the in-mem MMR.